lib/include/prjxray/xilinx/xc7series/ecc.h

```diff
@@ -8,36 +8,35 @@ namespace xilinx {
 namespace xc7series {
 
 // Extend the current ECC code with one data word (32 bit) at a given
-// index in the configuration packet and return the new ECC code.
+// word index in the configuration frame and return the new ECC code.
 
 uint32_t icap_ecc(uint32_t idx, uint32_t data, uint32_t ecc) {
-	uint32_t off = idx * 32;
+	uint32_t val = idx * 32;  // bit offset
 
-	if (idx > 0x25)			// avoid 0x800
-		off += 0x1360;
-	else if (idx > 0x6)		// avoid 0x400
-		off += 0x1340;
-	else				// avoid lower
-		off += 0x1320;
+	if (idx > 0x25)  // avoid 0x800
```

Comment:
Not required in this PR but I'd like to see a writeup on the ECC calculation including why these special cases exist.